### PR TITLE
Refactor parse activity

### PR DESCRIPTION
**Pull request by Shine-neko**

### CODE REVIEW

File: src/Tempo/Bundle/AppBundle/Controller/ProviderActivityController.php [hunk 1]
```diff
@@ -15,31 +15,40 @@
...
      * @return Response
      * @throws \Exception
      */
-    public function providerAction(Request $request, $id)
+    public function providerAction(Request $request, Project $project, $provider)
     {
-        $manager = $this->getManager('activity_provider');
-        $projectProvider = $this->getManager('project_provider')->find($id);
-        $provider = $this->getProvider(strtoupper($projectProvider->getName()));
+        if (!$project->getMemberByUser($this->getUser())) {
```
Comment (amenophis) [inline src/Tempo/Bundle/AppBundle/Controller/ProviderActivityController.php]: Use a Voter ?

File: src/Tempo/Bundle/AppBundle/Resources/config/routing/provider_api.yml [hunk 2]
```diff
@@ -1,6 +1,7 @@
-api_provider_activity_ws:
-    path: /notify/{id}
+api_provider_activity_notify:
+    path: /{project}/notify/{provider}
     defaults:
         _controller: TempoAppBundle:ProviderActivity:provider
+        provider : default
     requirements:
           slug: "\w+\/\w+"
```
Comment (amenophis) [inline src/Tempo/Bundle/AppBundle/Resources/config/routing/provider_api.yml]: There is no slug in the path ?

Comment (Shine-neko) [inline src/Tempo/Bundle/AppBundle/Resources/config/routing/provider_api.yml]: Nop

File: src/Tempo/Bundle/AppBundle/Controller/ProviderActivityController.php (hunk 1 above)
Comment (Shine-neko) [inline src/Tempo/Bundle/AppBundle/Controller/ProviderActivityController.php]: why ?

Comment (Shine-neko) [inline src/Tempo/Bundle/AppBundle/Controller/ProviderActivityController.php]: Pas faux :D

Comment (amenophis) [inline src/Tempo/Bundle/AppBundle/Controller/ProviderActivityController.php]: I always say the truth !

File: src/Tempo/Bundle/AppBundle/Resources/config/routing/provider_api.yml (hunk 2 above)
Comment (amenophis) [inline src/Tempo/Bundle/AppBundle/Resources/config/routing/provider_api.yml]: So why a requirement ?

File: src/Tempo/Bundle/AppBundle/Manager/ActivityManager.php [hunk 3]
```diff
@@ -48,10 +48,11 @@ public function build($target, $action, $data, $author = null)
     }
 
     /**
+     * @param Project|null $parent
      * @param $user
      */
-    public function getActivities($user)
+    public function getActivities($project, $user)
```
Comment (amenophis) [inline src/Tempo/Bundle/AppBundle/Manager/ActivityManager.php]: TypeHint ?

File: src/Tempo/Bundle/AppBundle/Manager/ActivityProviderManager.php [hunk 4]
```diff
@@ -30,9 +30,9 @@ public function addActivity(ActivityProviderInterface $activity, ProjectProvider
         return $activity;
     }
 
-    public function getActivities($user)
+    public function getActivities($project, $user)
```
Comment (amenophis) [inline src/Tempo/Bundle/AppBundle/Manager/ActivityProviderManager.php]: TypeHint

File: src/Tempo/Bundle/AppBundle/Repository/ActivityProviderRepository.php [hunk 5]
```diff
@@ -20,15 +20,23 @@
  */
 class ActivityProviderRepository extends EntityRepository
 {
-    public function findUserActivites($user)
+    public function findActivities($project, $user, $provider = null)
```
Comment (amenophis) [inline src/Tempo/Bundle/AppBundle/Repository/ActivityProviderRepository.php]: TypeHint ?

File: src/Tempo/Bundle/AppBundle/Repository/ActivityProviderRepository.php [hunk 6]
```diff
@@ -20,15 +20,23 @@
...
         $query = $this->createQueryBuilder('activity');
         $query
             ->leftJoin('activity.provider', 'provider')
-            ->leftJoin('provider.project', 'project')
-            ->leftJoin('project.members', 'user')
-            ->andWhere('user.user = :user')
-            ->setParameter('user', $user);
+            ->leftJoin('provider.project', 'project');
+
+        if(null === $project ) {
```
Comment (amenophis) [inline src/Tempo/Bundle/AppBundle/Repository/ActivityProviderRepository.php]: bad space

File: src/Tempo/Bundle/AppBundle/Repository/ActivityRepository.php [hunk 7]
```diff
@@ -25,11 +26,25 @@ public function getUserActivites($user)
...
-            ->where('activity.target IN(:target)')
-            ->andWhere('user.user = :user')
-            ->setParameter('target', array('Project', 'Comment'))
+            ->where('user.user = :user')
             ->setParameter('user', $user);
 
+        return $query;
+    }
+
+    public function getProjectActivities($parent, $user)
```
Comment (amenophis) [inline src/Tempo/Bundle/AppBundle/Repository/ActivityRepository.php]: TypeHint ?

File: src/Tempo/Bundle/AppBundle/Resources/config/routing/provider_api.yml [hunk 8]
```diff
@@ -1,6 +1,5 @@
-api_provider_activity_ws:
-    path: /notify/{id}
+api_provider_activity_notify:
+    path: /{project}/notify/{provider}
     defaults:
-        _controller: TempoAppBundle:ProviderActivity:provider
-    requirements:
-          slug: "\w+\/\w+"
+        _controller: TempoAppBundle:Provider:notifiy
```
Comment (amenophis) [inline src/Tempo/Bundle/AppBundle/Resources/config/routing/provider_api.yml]: Why not in defaults ?

File: src/Tempo/Bundle/AppBundle/Resources/views/Provider/list.html.twig [hunk 9]
```diff
@@ -11,9 +11,9 @@
                 </tr>
             </thead>
             <tbody>
-                {% for provider in providers  %}
+                {% for provider in providers %}
                 <tr>
-                    <td><span class="glyphicon glyphicon-off clgray"></span></td>
+                    <td> <span class="glyphicon glyphicon-off clgray"></span></td>
```
Comment (amenophis) [inline src/Tempo/Bundle/AppBundle/Resources/views/Provider/list.html.twig]: bad space

**Pull request merged**